### DISCUSSION TITLE
Which test runner is preferred ? Jest or Mocha

### DISCUSSION THREAD

**Discussion by ShivamJoker** [wix/Detox#3059] [Q&A] [answered]
In the getting started guide I saw Mocha was the preferred runner but in another issue I saw @d4vidi mentioned that wix has now moved to Jest.

It would be really helpful to get some insights on this.

If Jest should be the go to runner for new people then please update the docs.

**Comment by noomorph** [comment, marked as answer]
Okay, I'll check the docs, but I'd appreciate it if you point me to the outdated fragment. Thanks in advance!

![image](https://user-images.githubusercontent.com/1962469/142165964-6e9f14bd-948c-46d2-a6f9-480123346452.png)

In `Getting Started`, at least we have it stated clearly.

**Reply by ShivamJoker** [reply]
Oh maybe I misspoke, but somewhere I saw that at wix you guys were using mocha only.

But after looking at jest docs again it looks bit complicated to me and also while generating the template even with mocha the beforeAll statement comes instead of just before.

So should I consider mocha doesn't have the same amount of support or users ?

**Reply by noomorph** [reply]
> So should I consider mocha doesn't have the same amount of support

Correct. There are quite a few important features that are unavailable for Mocha.

> users?

To be honest, we've never tried to count Detox userbase between both platforms (Mocha vs Jest), but I'd assume that Jest is a more popular choice for Detox by far.

As for the runners themselves, personally, I don't have any negative bias towards Mocha – at the time when we switched the recommendation, Mocha lacked a few features (e.g., async hooks for any test events) we counted on (e.g., for the screenshot artifacts). Now it's just that we don't have the capacity to give it another try – there are far more important updates we need to focus on (support multiple apps, multiple devices in the same test; artifacts improvements, etc.). This is why things are the way they are.

**Reply by ShivamJoker** [reply]
Alright thanks for the info
